expect.go

```diff
@@ -237,6 +238,50 @@ func (DefaultRequestFactory) NewRequest(
 	return http.NewRequest(method, urlStr, body)
 }
 
+// DefaultFormatter is the default Formatter implementation.
+type DefaultFormatter struct{}
+
+// BeginAssertion implements Formatter.BeginAssertion.
+//
+// It is a no-op for now. Actual implementation may be
+// added later.
+func (DefaultFormatter) BeginAssertion(ctx Context) {}
+
+// Success implements Formatter.Success.
+//
+// It is a no-op for now. Actual implementation may be
+// added later.
+func (DefaultFormatter) Success(ctx Context) {}
+
+// EndAssertion implements Formatter.EndAssertion.
+//
+// It is a no-op for now. Actual implementation may be
+// added later.
+func (DefaultFormatter) EndAssertion(ctx Context) {}
+
+// Failure implements Formatter.Failure and reports failure.
+//
+// It formats the info from Context and Failure struct into
+// a string and passes it to Context.Reporter for reporting.
+func (DefaultFormatter) Failure(ctx Context, f Failure) {
+	errString := ""
+	if f.actual != nil {
```

Comment:
I think we should also add assertionName to Failure struct (e.g. "Equals", "Contains", etc) and print it here.

Reply:
Also, I suggest to rename Context.name to Context.TestName. And I guess we need to make all Context fields, except formatter, exported.

Reply:
the test (`testing.T`) reference could be inside the context?